core/lib/request.py

```diff
@@ -25,11 +24,10 @@ def __init__(self, type, method, url, parent=None, referer=None, data=None, trig
                  set_cookie=None, http_auth=None, db_id=None, parent_db_id=None, out_of_scope=None):
         self.type = type
         self.method = method
-        self._html = None
-        self._html_hash = None
         self.user_output = []
+        self.html = None
+        self._html_hash = None
```

Comment:
we will need to store it in the db to ensure to not crawl it again on subsequent crawl.

Reply:
This is an issue if we do not clear the db each time (warden use case?)